src/aiortc/rtcpeerconnection.py

```diff
@@ -742,14 +742,27 @@ def getTransceivers(self) -> list[RTCRtpTransceiver]:
         return list(self.__transceivers)
 
     async def setLocalDescription(
-        self, sessionDescription: RTCSessionDescription
+        self, sessionDescription: Optional[RTCSessionDescription] = None
     ) -> None:
         """
         Change the local description associated with the connection.
 
         :param sessionDescription: An :class:`RTCSessionDescription` generated
                                     by :meth:`createOffer` or :meth:`createAnswer()`.
+                                    or None to implicitly create one.
         """
+        if sessionDescription is None:
+            # https://w3c.github.io/webrtc-pc/#dom-peerconnection-setlocaldescription
+            # If left out, then setLocalDescription will implicitly create an offer
+            # or create an answer, as needed.
+            self.__log_debug(
+                "setLocalDescription()",
+            )
+            if self.signalingState == "have-remote-offer":
+                sessionDescription = await self.createAnswer()
+            else:
+                sessionDescription = await self.createOffer()
```

Comment:
It means we're generating an offer for:

- `closed`
- `have-local-offer`
- `have-local-pranswer`
- `have-remote-answer`
- `have-remote-pranswer`
- `stable`

Is this all correct?

Reply:
closed was missing a check (which I don't see in the spec either :thinking:)
For have-local-offer one might set the SDP to the current one but I do not see a difference (this is observable via the ice-ufrag which stays the same)

have-remote-(pr)anwer isn't a thing ;-)